server/util/approxlru/approxlru.go

```diff
@@ -230,20 +233,32 @@ func (l *LRU[T]) evict() (*Sample[T], error) {
 
 	for {
 		for i, sample := range l.samplePool {
-			log.Infof("Evictor attempting to evict %q", sample.Key)
+			l.mu.Lock()
+			oldLocalSizeBytes := l.localSizeBytes
+			oldGlobalSizeBytes := l.globalSizeBytes
+			l.mu.Unlock()
+
+			log.Infof("Evictor attempting to evict %q (last accessed %s)", sample.Key, time.Since(sample.Timestamp))
 			skip, err := l.onEvict(l.ctx, sample)
 			if err != nil {
 				log.Warningf("Could not evict %q: %s", sample.Key, err)
 				continue
 			}
 
 			l.mu.Lock()
-			l.localSizeBytes -= sample.SizeBytes
-			// Assume eviction on remote servers is happening at the same
-			// rate as local eviction. It's fine to be wrong as we expect the
-			// actual sizes to be periodically to be reset to the true numbers
-			// using UpdateSizeBytes from data received from other servers.
-			l.globalSizeBytes -= int64(float64(sample.SizeBytes) * float64(l.globalSizeBytes) / float64(l.localSizeBytes))
+			// Only update the size if it hasn't changed already. Users
+			// are allowed to update the size as a side-effect of the eviction
+			// callback which should reflect the new size.
+			if l.localSizeBytes == oldLocalSizeBytes {
+				l.localSizeBytes -= sample.SizeBytes
+			}
+			if l.globalSizeBytes == oldGlobalSizeBytes {
+				// Assume eviction on remote servers is happening at the same
```

Comment:
this comment might be clearer if you explained that:
 - pebble keeps its own total cache size and
 - pebble calls approxLRU.UpdateSize() with that new number periodically, so our tracking here can do this